services/visual_recognition/v3.js

```diff
@@ -62,301 +37,331 @@ module.exports = function (RED) {
+      }
+    } // for
+    async.parallel(asyncTasks, function(error, results){
+      if (error) {
+        //console.log('createClassifier ended with error ' + error);
+        throw error;
+      }
+      for (var p in listParams) {
+        if (!p) { // for codacy
+          params[p]=listParams[p];
         }
-      }; // actionComplete2
-
-      var actionCompleteDeleteClassifier = function(err, body, other) {
+      }
+      cb();
+    });
+  }
 
-        if (err != null && body==null)
-        {
-          node.status({fill:'red', shape:'ring', text:'call to watson visual recognition v3 service failed'}); 
-          msg.result = {};
-          msg.result['error_code']= err.code;
-          if (!err.error)
-            msg.result['error']= err.error;
-          console.log('Error:', err.code);
-          node.error('Error code : ' +  err.code);
-          return;
-        }
-        else {
-          msg.result = 'Deleted classifier_id';
-          msg.payload='see msg.result';
-          node.send(msg); 
-          node.status({});
-        }
-      }; // actionCompleteDeleteClassifier
-      
-      // If the input is an image, need to stream the input in, giving time for the
-      // data to arrive, before invoking the service. 
-      node.status({fill:'blue', shape:'dot', text:'Calling '+ feature + ' ...'});
-      if (imageCheck(msg.payload)) {
-        temp.open({suffix: '.' + fileType(msg.payload).ext}, function (err, info) {
-          if (err) {
-            this.status({fill:'red', shape:'ring', text:'unable to open image stream'});          
-            message ='Node has been unable to open the image stream'; 
-            node.error(message, msg);
-            return;        
-          }  
-          stream_buffer(info.path, msg.payload, function () {
-          params['images_file'] = fs.createReadStream(info.path);
-          if ( msg.params != null && msg.params.classifier_ids != null)
-            params['classifier_ids']=msg.params['classifier_ids'];
-          if ( msg.params != null && msg.params.owners != null)
-            params['owners']=msg.params['owners'];
-          if ( msg.params != null && msg.params.threshold != null)
-            params['threshold']=msg.params['threshold'];
-          performAction(params, feature, actionComplete);
+  function performDeleteAllClassifiers(params, node, msg) {
+    node.service.listClassifiers(params, function(err, body) {
+      node.status({});
+      if (err) {
+        node.status({fill:'red', shape:'ring', 
+          text:'Delete All : call to listClassifiers failed'});
+        node.error(err, msg);   
+      } else {
+        // Array to hold async tasks
+        var asyncTasks=[], nbTodelete=0, nbdeleted=0;
+        nbTodelete = body.classifiers.length;
+        body.classifiers.forEach(function (aClassifier) {
+          asyncTasks.push(function (cb) {
+            var parms = {};
+            parms.classifier_id=aClassifier.classifier_id;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Identifier 'classifier_id' is not in camel case.](https://www.codacy.com/app/BetaWorks-NodeRED-Watson/node-red-node-watson/file/2737047177/issues/source?bid=3330694&fileBranchId=3352492#l233)